src/sage/topology/moment_angle_complex.py

```diff
@@ -814,3 +819,712 @@ def has_trivial_lowest_deg_massey_product(self):
+
+        # We extract the cocycle monomials in
+        # order to loop over them
+        left_cocycles = left[2][1].monomials()
+        right_cocycles = right[2][1].monomials()
+        res = self.zero()
+        for left_cocycle in left_cocycles:
+            for right_cocycle in right_cocycles:
+                # We acquire the set of vertices of cocyles,
+                # because of the union
+                left_cocycle_vertices = left_cocycle.leading_support().set()
+                right_cocycle_vertices = right_cocycle.leading_support().set()
+                union = set_left.union(set_right)
+                res_union = left_cocycle_vertices.union(right_cocycle_vertices)
+                subcomplex_union = self._complex._simplicial_complex.generated_subcomplex(union, is_mutable=False)
+
+                # Because we special-cased the multiplicative identity
+                # at the beginning of the method, we know that both
+                # left[1] and right[1] are at least 0 (they represent
+                # the degree of the corresponding cochains). Therefore,
+                # deg is at least 1.
+                deg = left[1] + right[1] + 1
+                cochains_basis = subcomplex_union.n_chains(deg, cochains=True).basis()
+                # If the resulting cochain exists, then we
+                # add it to the result; otherwise, we add 0
+                if cochains_basis.has_key(Simplex(res_union)):
+                    res_part = self.zero()
+                    res_cochain = cochains_basis[Simplex(res_union)]
+                    # The algebraic_topological_model() works with unreduced cohomology,
+                    # but we require reduced cohomology here. This only causes problems
+                    # in dimension 0, but because of excluding the unit element at
+                    # the beginning of the method, we know that deg is at least 1
+                    # See comments at the beginning of the method and when defining deg
+                    phi, _ = subcomplex_union.algebraic_topological_model(self._base_ring)
+                    coeff_vec = phi.dual().pi().in_degree(deg) * res_cochain.to_vector()
+                    # We compute the sign of the result
+                    for i in range(len(coeff_vec)):
+                        res_part += coeff_vec[i] * self.basis()[li[0]+ri[0], i]
+                    zeta = 1
+                    for k in set_left.difference(left_cocycle_vertices):
+                        zeta *= eps({k}, set_right.union({k}).difference(right_cocycle_vertices))
+                    epsilon = (eps(left_cocycle_vertices, set_left)
+                               * eps(right_cocycle_vertices, set_right)
+                               * eps(res_union, union)
+                               * zeta)
+                    if epsilon == -1:
+                        res_part = -res_part
+                    res += res_part
+
+        return res
+
+    def cup_length(self):
```

Comment:
```suggestion
    @cached_method
    def cup_length(self, elements=False):
```
It would be good to have an option to return the list of elements that give the cup length.